vsphere/resource_vsphere_license_test.go

```diff
@@ -0,0 +1,193 @@
+package vsphere
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/license"
+)
+
+func TestAccVSphereLicenseBasic(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccVSphereLicenseDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseBasicConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseExists("vsphere_license.foo"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseInvalid(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseInvalidConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseNotExists("vsphere_license.foo"),
+				),
+				ExpectError: regexp.MustCompile("License file not found"),
+			},
+		},
+	})
+
+}
+
+func TestAccVSphereLicenseWithLabels(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccVSpherePreLicenseBasicCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVSphereLicenseWithLabelConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccVSphereLicenseWithLabelExists("vsphere_license.foo"),
+				),
+			},
+		},
+	})
+
+}
+
+func testAccVSphereLicenseInvalidConfig() string {
+
+	// quite sure this key cannot be valid
+	return `resource "vsphere_license" "foo" {
+  					license_key = "HN422-47193-58V7M-03086-0JAN2"
+			}`
+}
+
+func testAccVSphereLicenseWithLabelConfig() string {
+	// precheck already checks if this is present or not
+	key := os.Getenv("VSPHERE_LICENSE")
+	return fmt.Sprintf(`resource "vsphere_license" "foo" {
+							license_key = "%s"
+
+							labels {
+								VpxClientLicenseLabel = "Hello World"
+								TestTitle = FooBar
+							}		 	
+						}`, key)
+}
+
+func testAccVSphereLicenseBasicConfig() string {
+	// precheck already checks if this is present or not
+	key := os.Getenv("VSPHERE_LICENSE")
+	return fmt.Sprintf(`resource "vsphere_license" "foo" {
+  							license_key = "%s"
+						}
+						`, key)
+}
+
+func testAccVSphereLicenseDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*govmomi.Client)
```

Comment:
@vancluever So should I change it in all the places? Like other tests and methods.